### PR TITLE
feat(telemetry): add basic WAF telemetry metrics tests

### PR DESCRIPTION
**Pull request by avara1986**
## Description

Add basic WAF telemetry metrics tests

Python related PR: https://github.com/DataDog/dd-trace-py/pull/4943

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Few step for a new scenario : 

* update your branch with the last version of main
* add your scenario in `utils/scripts/generate-github-action-file.py` (you'll find in it a scenario list
* run this script (it'll update ci.yaml)

Thanks !

**Comment by cbeauchesne** [comment]
Finally, you want to keep a separated scenario? (cf discussion on slack)

**Review by gnufede — APPROVED** [review]

File: tests/appsec/waf/test_telemetry.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            else:
+                assert value is not None, f"Empty `{key}` header"
+
+    if len(expected_headers) > 0:
+        raise AssertionError(
+            "Headers %s not found in payload headers: %s" % ([header for header in expected_headers], headers)
+        )
+
+
+def _validate_metrics(payload, language):
```
Comment (cbeauchesne) [inline tests/appsec/waf/test_telemetry.py:34]: `language` is always equals to `context.library`

```suggestion
def _validate_metrics(payload):
```

File: tests/appsec/waf/test_telemetry.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+from utils import interfaces, released, rfc, weblog, coverage, scenario, context
+from utils.tools import logger
+
+TELEMETRY_REQUEST_TYPE_GENERATE_METRICS = "generate-metrics"
+
+
+def _validate_headers(headers, language):
```
Comment (cbeauchesne) [inline tests/appsec/waf/test_telemetry.py:7]: `language` is always equals to `context.library`

```suggestion
def _validate_metrics(payload):
```

File: tests/appsec/waf/test_telemetry.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                    assert data["response"]["status_code"] == 202
+                    _validate_headers(data["request"]["headers"], context.library)
+                    _validate_metrics(data["request"]["content"]["payload"], context.library)
+                    return True
+            except Exception as e:
+                logger.error(str(e), exc_info=True)
+                raise
+            return False
+
+        interfaces.library.validate_telemetry(validator, success_by_default=False)
```
Comment (cbeauchesne) [inline tests/appsec/waf/test_telemetry.py:86]: `success_by_default` default value is always `False`

```suggestion
        interfaces.library.validate_telemetry(validator)
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**